pkg/metrics/wal/series.go

```diff
@@ -84,6 +84,12 @@ func (m seriesHashmap) del(hash uint64, ref uint64) {
 	}
 }
 
+type memExemplar struct {
+	ts     int64
+	value  float64
+	labels labels.Labels
+}
```

Comment:
I think we should probably just use `exemplar.Exemplar` here instead of redefining it. That'd help clean up the code a bit (ie you can call `.Equals` for equality checking) and it also means you'd get back the HasTs checking, which seems to be potentially important.

Reply:
It's fine to import Prometheus packages 👍 We only duplicated memSeries because it was too complex to adapt it to support a mode that didn't create chunks on disk.

Reply:
I've changed this in 249600d